bitnami/spark/README.md

```diff
@@ -0,0 +1,256 @@
+| `service.loadBalancerIP`                    | loadBalancerIP if spark service type is `LoadBalancer`                                      | `nil`                                                   |
+| `ingress.enabled`                           | Enable the use of the ingress controller to access the web UI                               | `true`                                                  |
+| `ingress.hosts`                             | Add hosts to the ingress controller with name and path                                      | `name: spark.local`, `path: /`                          |
+
+
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install --name my-release \
+  --set master.webPort=8081 bitnami/spark
+```
+
+The above command sets the spark master web port to `8081`.
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```console
+$ helm install --name my-release -f values.yaml bitnami/spark
+```
+
+> **Tip**: You can use the default [values.yaml](values.yaml)
+
+### Production configuration and horizontal scaling
+
+This chart includes a `values-production.yaml` file where you can find some parameters oriented to production configuration in comparison to the regular `values.yaml`.
+
+```console
+$ helm install --name my-release -f ./values-production.yaml bitnami/spark
+```
+
+- Enable RPC authentication:
+```diff
+- security.rpc.authenticationEnabled: no
++ security.rpc.authenticationEnabled: yes
+```
+
+- Enable RPC encryption:
+```diff
+- security.rpc.encryptionEnabled: no
++ security.rpc.encryptionEnabled: yes
+```
+
+- Enable storage encryption:
+```diff
+- security.storageEncryptionEnabled: no
++ security.storageEncryptionEnabled: no
+```
+
+- Configure SSL parameters:
+```diff
+- security.ssl.enabled: no
+- security.ssl.needClientAuth: no
++ security.ssl.enabled: yes
++ security.ssl.needClientAuth: yes
+```
+
+- Set a secret name for passwords:
+```diff
++ security.passwordsSecretName: my-passwords-secret
+```
+
+- Set a secret name for certificates:
+```diff
++ security.certificatesSecretName: my-certificates-secret
+```
+
+- Enable authentication between master and workers:
+```diff
+- auth.enabled: false
++ auth.enabled: true
+```
+
+- Enable SSL communication:
+```diff
+- ssl.enabled: false
++ ssl.enabled: true
+```
+
+- Enable local storage encryption:
+```diff
+- storageEncription.enabled: false
++ storageEncription.enabled: true
+```
+
+### How to create the certificates secret
+It is needed to create two secrets to set the passwords and certificates. The name of the two secrets should be configured on `security.passwordsSecretName` and `security.certificatesSecretName`.
+Into the certificates secret, the keys must be `spark-keystore.jks` and `spark-truststore.jks`, and the content must be text on JKS format.
```

Comment:
Any link on how to generate a jks certificate and keystore?

Reply:
OKay.